apps/rush-lib/src/logic/versionMismatch/VersionMismatchFinder.ts

```diff
@@ -96,20 +112,25 @@ export class VersionMismatchFinder {
     options: {
       isRushCheckCommand: boolean;
       variant?: string | undefined;
+      jsonFlag?: boolean | undefined;
     }
   ): void {
 
     if (rushConfiguration.ensureConsistentVersions || options.isRushCheckCommand) {
       const mismatchFinder: VersionMismatchFinder = VersionMismatchFinder.getMismatches(rushConfiguration, options);
 
-      mismatchFinder.print();
-
-      if (mismatchFinder.numberOfMismatches) {
-        console.log(colors.red(`Found ${mismatchFinder.numberOfMismatches} mis-matching dependencies!`));
-        process.exit(1);
+      if (options.jsonFlag) {
+        console.log(JSON.stringify(mismatchFinder.getMismatchJson(), undefined, 2));
```

Comment:
We already have `mismatchFinder.print();`. Can we instead create a function on `VersionMismatchFinder` called `printAsJson` that does the printing as well?